src/stories/Forms/Select/Select.tsx

```diff
@@ -0,0 +1,489 @@
+
+  useEffect(() => {
+    setSelected(value);
+  }, [value]);
+
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      const selectList = selectListRef.current;
+      const selectListChildren = selectList?.children as HTMLCollectionOf<HTMLElement>;
+      if (!selectListChildren) return;
+
+      const activeElement = document.activeElement as HTMLElement;
+      const activeElementIndex = Array.from(selectListChildren).indexOf(activeElement);
+
+      if (['Escape', 'ArrowDown', 'ArrowUp', 'Enter', ' '].includes(e.key)) {
+        e.preventDefault();
+      }
+
+      switch (e.key) {
+        case ' ':
+          if (activeElement === selectButtonRef.current && !isOpen) {
+            setIsOpen(true);
+          }
+          break;
+
+        case 'Escape':
+          setIsOpen(false);
+          selectButtonRef?.current?.focus();
+          break;
+
+        case 'ArrowUp':
+          if (activeElementIndex === -1) return;
+          if (activeElementIndex === 0) {
+            selectList.focus();
+          } else {
+            selectListChildren[activeElementIndex - 1].focus();
+          }
+          break;
+
+        case 'ArrowDown':
+          if (activeElementIndex === -1) return;
+          if (activeElementIndex === selectListChildren.length - 1) {
+            selectList.focus();
+          } else {
+            selectListChildren[activeElementIndex + 1].focus();
+          }
+          break;
+
+        case 'Enter':
+          if (activeElementIndex !== -1) {
+            selectListChildren[activeElementIndex].click();
+          }
+          break;
+        case 'Tab':
+          filterSearchRef?.current?.setAttribute('tabindex', activeElement === selectButtonRef.current ? '-1' : '0');
+          break;
+        default:
+          if (!/^[a-zA-Z0-9]$/.test(e.key) || !isOpen) return;
+          if (filterSearchRef.current) {
+            filterSearchRef.current.focus();
+          }
+          break;
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen]);
+
+  useEffect(() => {
+    setSelectListHeight(getSelectListHeight(showItems, selectListRef));
+    handleSelectHoverBackground(selectListRef, selectListHoverBackgroundRef);
+  }, [selectListRef, children, filterSearch]);
+
+  useOnClickOutside(selectWrapperRef, () => setIsOpen(false));
+
+  const handleValueSelect = (event) => {
+    const { target } = event;
+    const selectedValue = target.dataset.value;
+    if (!selectedValue) {
+      console.error('Select component: value is not defined', target);
+      return;
+    }
+    setFilterSearch('');
+    setIsOpen(false);
+    setSelected(selectedValue);
+    selectButtonRef?.current?.focus();
+    onChange &&
+      onChange({ target: { name: selectName, id, value: selectedValue } } as React.ChangeEvent<HTMLInputElement>);
```

Comment:
Note to self, discuss onChange events with Rosti

Reply:
As discussed: Returning a fake event target so the value is compatible with Formik.